DDECal/TECConstraint.cc

```diff
@@ -53,8 +54,21 @@ void ApproximateTECConstraint::initializeChild()
 
 void TECConstraintBase::applyReferenceAntenna(std::vector<std::vector<dcomplex> >& solutions) const
 {
-  // TODO chose this more cleverly?
+  // Choose reference antenna that has at least 20% channels unflagged
   size_t refAntenna = 0;
+  for(; refAntenna!=_nAntennas; ++refAntenna)
+  {
+    size_t nUnFlaggedChannels=0;
+    // Only check flagged state for first direction
+    for (size_t ch=0; ch!=_nChannelBlocks; ++ch)
+    {
+      if(isfinite(solutions[0][refAntenna*_nDirections]))
+        nUnFlaggedChannels++;
+    }
+    if (nUnFlaggedChannels*1.0/_nChannelBlocks > 0.2)
+      // Choose this refAntenna;
+      break;
+  }
```

Comment:
Add a check:
```
if(refAntenna == _nAntennas)
  throw std::runtime_error("Could not found a reference antenna: all antennas appear to be flagged");
```

Reply:
I don't think this is a good idea. When one timeslot is fully flagged, we don't want DPPP to crash on a runtime error.

Reply:
Ah that's right, I forgot this is per solution interval. But an `if` is required nevertheless, because coming out with `refant=nAnt` will cause a segfault. So then it should just do `refant=0` in the case that no antenna is found.

Reply:
Good point, I did that.